dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/enrollment/EnrollmentService.java

```diff
@@ -58,6 +59,15 @@ List<Enrollment> getEnrollments(EnrollmentOperationParams params)
   Page<Enrollment> getEnrollments(EnrollmentOperationParams params, PageParams pageParams)
       throws BadRequestException, ForbiddenException;
 
+  /**
+   * Returns a list of existing Enrollments from the provided UIDs
+   *
+   * @param uids Enrollment UIDs to check
+   * @return Enrollment list
+   */
+  List<Enrollment> getEnrollments(@Nonnull List<String> uids, UserDetails currentUser)
```

Comment:
```suggestion
  List<Enrollment> getEnrollments(@Nonnull List<String> uids, UserDetails user)
```

Can you transition it to `List<UID>` please.

Regarding the javadoc: I would remove the `@param` and `@return` unless we have something useful to say about them. I would mention the user as in https://github.com/dhis2/dhis2-core/blob/79cb6c1eda5b3c22dab9862ff251e02ce2a8585d/dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventService.java#L68-L72

Reply:
I agree we should use a `List<UID>` but to do so we need to validate the uids that are strings in the `DeduplicationController`. We can either explicitly validate them or we can change the `MergeObject` to use `List<UID>` instead of `List<String>`.

In general I agree to use `UID` wherever it makes sense, and this means that if we validated a string to be a `UID` than it makes sense to never go back to string.